packages/storage/amplify_storage_s3_android/android/src/main/kotlin/com/amazonaws/amplify/amplify_storage_s3/StorageS3.kt

```diff
@@ -45,23 +47,39 @@ class StorageS3 : FlutterPlugin, ActivityAware, MethodCallHandler {
 
     override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
         channel =
-            MethodChannel(flutterPluginBinding.binaryMessenger, "com.amazonaws.amplify/storage_s3")
+                MethodChannel(flutterPluginBinding.binaryMessenger, "com.amazonaws.amplify/storage_s3")
         channel.setMethodCallHandler(this)
         context = flutterPluginBinding.applicationContext
 
         transferProgressEventChannel = EventChannel(
-            flutterPluginBinding.binaryMessenger,
-            "com.amazonaws.amplify/storage_transfer_progress_events"
+                flutterPluginBinding.binaryMessenger,
+                "com.amazonaws.amplify/storage_transfer_progress_events"
         )
         transferProgressEventChannel.setStreamHandler(transferProgressStreamHandler)
     }
 
     override fun onMethodCall(@NonNull call: MethodCall, @NonNull _result: Result) {
         val result = AtomicResult(_result, call.method)
 
-        if (call.method == "addPlugin") {
+        val arguments = call.arguments as Map<String, *>
+
+        if (call.method == "configureStorage") {
             try {
-                Amplify.addPlugin(AWSS3StoragePlugin())
+
+                var hasPrefixResolver = arguments["hasPrefixResolver"] as? Boolean? == true
+
+                if (hasPrefixResolver) {
+                    Amplify.addPlugin(
+                            AWSS3StoragePlugin(
+                                    AWSS3StoragePluginConfiguration {
+                                        awsS3PluginPrefixResolver = FlutterPrefixResolver(methodChannel = channel);
+                                    }
+                            )
+                    )
+                } else {
```

Comment:
nit: might not need the `if else` branch here as `awsS3PluginPrefixResolver` of `AWSS3StoragePluginConfiguration` is nullable.